File: Python/Easy/Filter/Sorting/2363__Merge_Similar_Items.py
from collections import Counter, defaultdict


def mergeSimilarItems(items1: list[list[int]], items2: list[list[int]]) -> list[list[int]]:
    value_weight_map = defaultdict(int)

    for value, weight in items1:
        value_weight_map[value] += weight

    for value, weight in items2:
        value_weight_map[value] += weight

    return [[value, weight] for value, weight in sorted(value_weight_map.items())]



items1 = [[1, 1], [4, 5], [3, 8]]
items2 = [[3, 1], [1, 5]]
print(mergeSimilarItems(items1, items2))

File: PHP/Easy/Filters/Sorting/2363__Merge_Similar_Items.php
<?php
function mergeSimilarItems($items1, $items2)
{
    $valueWeightMap = [];

    foreach ($items1 as $item) {
        $value = $item[0];
        $weight = $item[1];

        if (!isset($valueWeightMap[$value])) {
            $valueWeightMap[$value] = 0;
        }
        $valueWeightMap[$value] += $weight;
    }

    foreach ($items2 as $item) {
        $value = $item[0];
        $weight = $item[1];

        if (!isset($valueWeightMap[$value])) {
            $valueWeightMap[$value] = 0;
        }
        $valueWeightMap[$value] += $weight;
    }

    ksort($valueWeightMap);

    $result = [];
    foreach ($valueWeightMap as $value => $weight) {
        $result[] = [$value, $weight];
    }

    return $result;
}

$items1 = [[1, 1], [4, 5], [3, 8]];
$items2 = [[3, 1], [1, 5]];

print_r(mergeSimilarItems($items1, $items2));